src/phongo_apm.c

```diff
@@ -262,6 +262,35 @@ static void phongo_apm_command_failed(const mongoc_apm_command_failed_t* event)
 	FREE_HASHTABLE(subscribers);
 }
 
+static void phongo_apm_server_opening(const mongoc_apm_server_opening_t* event)
+{
+	mongoc_client_t*                 client;
+	HashTable*                       subscribers;
+	php_phongo_serveropeningevent_t* p_event;
+	zval                             z_event;
+
+	client      = mongoc_apm_server_opening_get_context(event);
+	subscribers = phongo_apm_get_subscribers_to_notify(php_phongo_sdamsubscriber_ce, client);
+
+	/* Return early if there are no APM subscribers to notify */
+	if (zend_hash_num_elements(subscribers) == 0) {
+		goto cleanup;
+	}
+
+	object_init_ex(&z_event, php_phongo_serveropeningevent_ce);
+	p_event = Z_SERVEROPENINGEVENT_OBJ_P(&z_event);
+
+	mongoc_apm_server_opening_get_topology_id(event, &p_event->topology_id);
+	p_event->host = mongoc_apm_server_opening_get_host(event);
```

Comment:
Note that the documentation for [`mongoc_apm_server_opening_get_host`](http://mongoc.org/libmongoc/1.17.5/mongoc_apm_server_opening_get_host.html) states that the return value is only valid for the lifetime of the `mongoc_apm_server_opening_t` event. Storing the returned pointer directly on the `php_phongo_serveropeningevent_t` struct is likely to lead to a segfault down the line when libmongoc eventually frees this value and PHPC retains a pointer to it.